pip/vcs/git.py

```diff
@@ -171,6 +179,12 @@ def get_revision(self, location):
             ['rev-parse', 'HEAD'], show_stdout=False, cwd=location)
         return current_rev.strip()
 
+    def get_remote_revision(self, url, ref):
```

Comment:
Since it is not part of `VersionControl` "API", I'd rather it be called `_get_remote_revision`